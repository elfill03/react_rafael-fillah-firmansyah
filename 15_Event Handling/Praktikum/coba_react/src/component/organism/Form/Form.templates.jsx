import React from "react";
import {
  ProductCategory,
  ProductDescription,
  ProductFreshness,
  ProductImage,
  ProductName,
  ProductPrice,
  Title,
} from "../../molecules";

const MainForm = () => {
  return (
    <>
      <div className="container">
        <div className="d-grid">
          <Title />
          <form className="needs-validation" autoComplete="off">
            <div
              className="d-grid m-auto has-validation"
              style={{ width: "min(100% - 10px, 40%)" }}
            >
              <ProductName />
              <ProductCategory />
              <ProductFreshness />
              <ProductImage />
              <ProductDescription />
              <ProductPrice />
              {/* <div
                className="alert alert-success mt-3"
                role="alert"
                id="alert"
                hidden=""
              /> */}
            </div>
          </form>
        </div>
      </div>
    </>
  );
};

export default MainForm;
