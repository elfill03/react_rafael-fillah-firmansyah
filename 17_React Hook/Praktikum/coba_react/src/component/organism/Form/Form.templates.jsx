import React, { useState } from "react";
import {
  ButtonCreateProduct,
  ButtonDelete,
  ButtonEdit,
  ButtonGenerateNumber,
} from "../../atoms";
import { ProductDescription, ProductImage, Title } from "../../molecules";

const Form = () => {
  const [productName, setProductName] = useState("");
  const [isValid, setIsValid] = useState(true);
  const [productCategory, setProductCategory] = useState("");
  const [productFreshness, setProductFreshness] = useState("");
  const [productPrice, setProductPrice] = useState("");
  const [products, setProducts] = useState([]);

  const handleInputChange = (event) => {
    const inputValue = event.target.value;
    if (inputValue.length <= 10) {
      setProductName(inputValue);
      setIsValid(true);
    } else {
      setIsValid(false);
    }
  };
  const inputClass = isValid ? "form-control" : "form-control is-invalid";
  const handleSubmit = (event) => {
    event.preventDefault();
    const product = {
      name: productName,
      category: productCategory,
      freshness: productFreshness,
      price: productPrice,
    };
    setProducts([...products, product]);
    setProductName("");
    setProductCategory("");
    setProductFreshness("");
    setProductPrice("");
  };

  return (
    <>
      <div className="container">
        <div className="d-grid">
          <Title />
          <form
            className="needs-validation"
            autoComplete="off"
            onSubmit={handleSubmit}
          >
            <div
              className="d-grid m-auto has-validation"
              style={{ width: "min(100% - 10px, 40%)" }}
            >
              <ButtonGenerateNumber />
              <h1 className="fs-3 fw-bold">Detail Product</h1>
              <div>
                <label htmlFor="productname" className="form-label">
                  Product Name
                </label>
                <input
                  type="text"
                  className={inputClass}
                  name="productname"
                  id="productname"
                  aria-describedby="productname"
                  style={{ width: "min(100% - 10px, 60%)" }}
                  autoFocus=""
                  value={productName}
                  onChange={handleInputChange}
                />
                {!isValid && (
                  <div className="invalid-feedback">
                    Product name must be 10 characters or less
                  </div>
                )}
              </div>
              {/* END PRODUCT NAME */}
              <div className="has-validation mt-3">
                <label htmlFor="productcategory" className="form-label">
                  Product Category :
                </label>
                <select
                  className="form-select mb-3"
                  style={{ width: "min(100% - 10px, 45%)" }}
                  value={productCategory}
                  onChange={(event) => setProductCategory(event.target.value)}
                >
                  <option hidden="">Choose...</option>
                  <option>Food</option>
                  <option>Fruit</option>
                  <option>Vegetable</option>
                </select>
              </div>
              {/* END PRODUCTCATEGORY */}
              <ProductImage />
              <div className="has-validation mt-3">
                <label className="form-label">Product Freshness :</label>
                <div className="form-check">
                  <input
                    className="form-check-input"
                    type="radio"
                    name="freshness"
                    id="freshness1"
                    value="Brand New"
                    checked={productFreshness === "Brand New"}
                    onChange={(event) =>
                      setProductFreshness(event.target.value)
                    }
                  />
                  <label className="form-check-label" htmlFor="freshness1">
                    Brand New
                  </label>
                </div>
                <div className="form-check">
                  <input
                    className="form-check-input"
                    type="radio"
                    name="freshness"
                    id="freshness2"
                    value="Second Hank"
                    checked={productFreshness === "Second Hank"}
                    onChange={(event) =>
                      setProductFreshness(event.target.value)
                    }
                  />
                  <label className="form-check-label" htmlFor="freshness2">
                    Second Hank
                  </label>
                </div>
                <div className="form-check">
                  <input
                    className="form-check-input"
                    type="radio"
                    name="freshness"
                    id="freshness3"
                    value="Refurbished"
                    checked={productFreshness === "Refurbished"}
                    onChange={(event) =>
                      setProductFreshness(event.target.value)
                    }
                  />
                  <label className="form-check-label" htmlFor="freshness3">
                    Refurbished
                  </label>
                </div>
              </div>
              {/* END PRODUCTFRESHNESS */}
              <ProductDescription />
              <div className="has-validation mt-3">
                <label htmlFor="productprice" className="form-label">
                  Product Price :
                </label>
                <input
                  type="number"
                  className="form-control"
                  id="productprice"
                  name="productprice"
                  aria-describedby="productprice"
                  value={productPrice}
                  onChange={(event) => setProductPrice(event.target.value)}
                  placeholder="$ 0"
                />
              </div>
              {/* END PRODUCTPRICE */}
              <ButtonCreateProduct />
            </div>
          </form>
          <div className="container m-auto text-center">
            <h1 className="fw-bold fs-2">List Product</h1>
          </div>
          <div className="container-fluid mb-0">
            <table
              className="table table-light table-striped"
              id="tableData"
              style={{ width: "min(100% - 10px, 75%)" }}
            >
              <thead>
                <tr>
                  <th scope="col">No</th>
                  <th scope="col">Product Name</th>
                  <th scope="col">Product Category</th>
                  <th scope="col">Image of Product</th>
                  <th scope="col">Product Freshness</th>
                  <th scope="col">Additional Desciption</th>
                  <th scope="col">Product Price</th>
                  <th scope="col">Action</th>
                </tr>
              </thead>
              <tbody>
                {products.map((product, index) => (
                  <tr key={index}>
                    <td>{index + 1}</td>
                    <td>{product.name}</td>
                    <td>{product.category}</td>
                    <td></td>
                    <td>{product.freshness}</td>
                    <td></td>
                    <td>{product.price}$</td>
                    <td>
                      <ButtonDelete /> <ButtonEdit />
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </>
  );
};

export default Form;
