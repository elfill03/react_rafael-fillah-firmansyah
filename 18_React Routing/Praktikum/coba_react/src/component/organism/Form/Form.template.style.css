.navigation {
  border: none;
  margin-bottom: 20px;
  padding-top: 5px;
  padding-bottom: 5px;
  background-color: rgb(232, 250, 215);
}
.navigation:hover {
  background-color: rgb(196, 255, 140);
}
