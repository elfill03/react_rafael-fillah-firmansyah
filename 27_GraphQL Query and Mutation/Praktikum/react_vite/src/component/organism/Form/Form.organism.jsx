import axios from "axios";
import { useFormik } from "formik";
import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import * as Yup from "yup";
import { useProductListSelector } from "../../../Config/redux/ProductList/productListSelector";
import { productAction } from "../../../Config/redux/ProductList/productSlice";

import {
  ButtonCreateProduct,
  ButtonDelete,
  ButtonEdit,
  ButtonGenerateNumber,
} from "../../atoms";
import {
  CreateProductTitle,
  ProductCategory,
  ProductDescription,
  ProductFreshness,
  ProductImage,
  ProductName,
  ProductPrice,
} from "../../molecules";
import "./Form.organism.style.css";

const Schema = Yup.object().shape({
  productname: Yup.string().min(2).max(20).required("Please fill this field!"),
  productcategory: Yup.string().required("Please choose Product Category!"),
  productimage: Yup.mixed().required("Please insert Product Image!"),
  productfreshness: Yup.string().required("Please choose Product Freshness!"),
  productdescription: Yup.string()
    .min(5)
    .max(150)
    .required("Please fill this fdescription!"),
  productprice: Yup.number().positive().required("Please fill the price!"),
});

const Form = ({ onNavigate }) => {
  const productList = useProductListSelector();
  const dispatch = useDispatch();

  const formik = useFormik({
    initialValues: {
      productname: "",
      productcategory: "",
      productimage: "",
      productfreshness: "",
      productdescription: "",
      productprice: 0,
    },
    validationSchema: Schema,
    onSubmit: (values, { resetForm }) => {
      axios
        .post(
          "https://643b9a2070ea0e660297037c.mockapi.io/products/products",
          values
        )
        .then((response) => {
          dispatch(productAction.add([...productList, response.data]));
          resetForm();
        })
        .catch((error) => {
          console.log(error);
        });
    },
  });

  const deleteProduct = (index) => {
    const isConfirmed = window.confirm(
      "Are you sure you want to delete this product?"
    );
    if (isConfirmed) {
      const productToDelete = productList[index];
      axios
        .delete(
          `https://643b9a2070ea0e660297037c.mockapi.io/products/products/${productToDelete.id}`
        )
        .then(() => {
          const updatedProducts = [...productList];
          updatedProducts.splice(index, 1);
          dispatch(productAction.delete([...updatedProducts]));
        })
        .catch((error) => {
          console.log(error);
        });
    }
  };

  useEffect(() => {
    axios
      .get("https://643b9a2070ea0e660297037c.mockapi.io/products/products")
      .then((response) => {
        dispatch(productAction.set(response.data));
      })
      .catch((error) => {
        console.log(error);
      });
  }, []);

  return (
    <>
      <div className="container">
        <div className="d-grid">
          <CreateProductTitle />
          <form
            id="createForm"
            className="needs-validation"
            autoComplete="off"
            onSubmit={formik.handleSubmit}
          >
            <div
              className="d-grid m-auto has-validation"
              style={{ width: "min(100% - 10px, 40%)" }}
            >
              <ButtonGenerateNumber />
              {/* <button onClick={onNavigate} className="navigation">
                Move To Landing Page
              </button> */}
              <ProductName
                onChange={formik.handleChange}
                value={formik.values.productname}
              />
              {formik.errors.productname && (
                <div className="error-msg">{formik.errors.productname}</div>
              )}

              <ProductCategory
                onChange={formik.handleChange}
                value={formik.values.productcategory}
                data-testid="productcategory-input"
              />
              {formik.errors.productcategory && (
                <div className="error-msg">{formik.errors.productcategory}</div>
              )}

              <ProductImage
                onChange={formik.handleChange}
                value={formik.values.productimage}
              />
              {formik.errors.productimage && (
                <div className="error-msg">{formik.errors.productimage}</div>
              )}

              <ProductFreshness
                checked={formik.values.productfreshness}
                onChange={formik.handleChange}
                value={formik.values.productfreshness}
              />
              {formik.errors.productfreshness && (
                <div className="error-msg">
                  {formik.errors.productfreshness}
                </div>
              )}

              <ProductDescription
                onChange={formik.handleChange}
                value={formik.values.productdescription}
              />
              {formik.errors.productdescription && (
                <div className="error-msg">
                  {formik.errors.productdescription}
                </div>
              )}

              <ProductPrice
                onChange={formik.handleChange}
                value={formik.values.productprice}
              />
              {formik.errors.productprice && (
                <div className="error-msg">{formik.errors.productprice}</div>
              )}

              <ButtonCreateProduct onClick={formik.handleSubmit} />
            </div>
          </form>
          <div className="container m-auto text-center">
            <h1 className="fw-bold fs-2">List Product</h1>
          </div>
          <div className="container-fluid mb-0">
            <table className="table table-light table-striped" id="tableData">
              <thead>
                <tr>
                  <th scope="col">No</th>
                  <th scope="col">Product Name</th>
                  <th scope="col">Product Category</th>
                  <th scope="col">Image of Product</th>
                  <th scope="col">Product Freshness</th>
                  <th scope="col">Additional Desciption</th>
                  <th scope="col">Product Price</th>
                  <th scope="col">Action</th>
                </tr>
              </thead>
              <tbody>
                {productList.map((values, index) => (
                  <tr key={index}>
                    <td>{index + 1}</td>
                    <td>{values.productname}</td>
                    <td>{values.productcategory}</td>
                    <td>{values.productimage}</td>
                    <td>{values.productfreshness}</td>
                    <td>{values.productdescription}</td>
                    <td>{values.productprice}$</td>
                    <td>
                      <ButtonDelete
                        onDelete={() => {
                          deleteProduct(index);
                        }}
                      />
                      <ButtonEdit />
                    </td>
                  </tr>
                ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </>
  );
};

export default Form;
