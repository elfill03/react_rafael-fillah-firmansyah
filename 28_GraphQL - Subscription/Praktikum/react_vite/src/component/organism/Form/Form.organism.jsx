import { gql, useMutation, useSubscription } from "@apollo/client";
import { useFormik } from "formik";
import React, { useEffect } from "react";
import { useDispatch } from "react-redux";
import * as Yup from "yup";
import { useProductListSelector } from "../../../Config/redux/ProductList/productListSelector";
import { productAction } from "../../../Config/redux/ProductList/productSlice";
import {
  ButtonCreateProduct,
  ButtonDelete,
  ButtonEdit,
  ButtonGenerateNumber,
} from "../../atoms";
import {
  CreateProductTitle,
  ProductCategory,
  ProductDescription,
  ProductFreshness,
  ProductImage,
  ProductName,
  ProductPrice,
} from "../../molecules";
import "./Form.organism.style.css";

const GetProductlist = gql`
  query MyQuery {
    List_products {
      id
      productcategory
      productdescription
      productfreshness
      productimage
      productname
      productprice
    }
  }
`;

const SUBSCRIBE_TO_PRODUCTS = gql`
  subscription {
    List_products {
      id
      productcategory
      productdescription
      productfreshness
      productimage
      productname
      productprice
    }
  }
`;

const CREATE_PRODUCT_MUTATION = gql`
  mutation CreateProduct($product: List_products_insert_input!) {
    insert_List_products_one(object: $product) {
      id
      productname
      productcategory
      productimage
      productfreshness
      productdescription
      productprice
    }
  }
`;

const DELETE_PRODUCT_MUTATION = gql`
  mutation DeleteProduct($id: Int!) {
    delete_List_products_by_pk(id: $id) {
      id
    }
  }
`;

const Schema = Yup.object().shape({
  productname: Yup.string().min(2).max(20).required("Please fill this field!"),
  productcategory: Yup.string().required("Please choose Product Category!"),
  productimage: Yup.mixed().required("Please insert Product Image!"),
  productfreshness: Yup.string().required("Please choose Product Freshness!"),
  productdescription: Yup.string()
    .min(5)
    .max(150)
    .required("Please fill this description!"),
  productprice: Yup.number().positive().required("Please fill the price!"),
});

const Form = ({ onNavigate }) => {
  const productList = useProductListSelector();
  const dispatch = useDispatch();

  const { loading, error, data } = useSubscription(SUBSCRIBE_TO_PRODUCTS);

  useEffect(() => {
    if (data) {
      dispatch(productAction.set(data.List_products));
    }
  }, [data, dispatch]);

  const [deleteProduct] = useMutation(DELETE_PRODUCT_MUTATION);

  const [createProduct] = useMutation(CREATE_PRODUCT_MUTATION);

  const formik = useFormik({
    initialValues: {
      productname: "",
      productcategory: "",
      productimage: "",
      productfreshness: "",
      productdescription: "",
      productprice: 0,
    },
    validationSchema: Schema,
    onSubmit: async (values, { resetForm }) => {
      try {
        await createProduct({ variables: { product: values } });
        resetForm();
      } catch (error) {
        console.log(error);
      }
    },
  });

  const handleDeleteProduct = (productId) => {
    const isConfirmed = window.confirm(
      "Are you sure you want to delete this product?"
    );
    if (isConfirmed) {
      deleteProduct({ variables: { id: productId } }).catch((error) => {
        console.log(error);
      });
    }
  };

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return <div>Error: {error.message}</div>;
  }

  return (
    <>
      <div className="container">
        <div className="d-grid">
          <CreateProductTitle />
          <form
            id="createForm"
            className="needs-validation"
            autoComplete="off"
            onSubmit={formik.handleSubmit}
          >
            <div
              className="d-grid m-auto has-validation"
              style={{ width: "min(100% - 10px, 40%)" }}
            >
              <ButtonGenerateNumber />
              {/* <button onClick={onNavigate} className="navigation">
                Move To Landing Page
              </button> */}
              <ProductName
                onChange={formik.handleChange}
                value={formik.values.productname}
              />
              {formik.errors.productname && (
                <div className="error-msg">{formik.errors.productname}</div>
              )}

              <ProductCategory
                onChange={formik.handleChange}
                value={formik.values.productcategory}
                data-testid="productcategory-input"
              />
              {formik.errors.productcategory && (
                <div className="error-msg">{formik.errors.productcategory}</div>
              )}

              <ProductImage
                onChange={formik.handleChange}
                value={formik.values.productimage}
              />
              {formik.errors.productimage && (
                <div className="error-msg">{formik.errors.productimage}</div>
              )}

              <ProductFreshness
                checked={formik.values.productfreshness}
                onChange={formik.handleChange}
                value={formik.values.productfreshness}
              />
              {formik.errors.productfreshness && (
                <div className="error-msg">
                  {formik.errors.productfreshness}
                </div>
              )}

              <ProductDescription
                onChange={formik.handleChange}
                value={formik.values.productdescription}
              />
              {formik.errors.productdescription && (
                <div className="error-msg">
                  {formik.errors.productdescription}
                </div>
              )}

              <ProductPrice
                onChange={formik.handleChange}
                value={formik.values.productprice}
              />
              {formik.errors.productprice && (
                <div className="error-msg">{formik.errors.productprice}</div>
              )}

              <ButtonCreateProduct onClick={formik.handleSubmit} />
            </div>
          </form>
          <div className="container m-auto text-center">
            <h1 className="fw-bold fs-2">List Product</h1>
          </div>
          <div className="container-fluid mb-0">
            <table className="table table-light table-striped" id="tableData">
              <thead>
                <tr>
                  <th scope="col">No</th>
                  <th scope="col">Product Name</th>
                  <th scope="col">Product Category</th>
                  <th scope="col">Image of Product</th>
                  <th scope="col">Product Freshness</th>
                  <th scope="col">Additional Description</th>
                  <th scope="col">Product Price</th>
                  <th scope="col">Action</th>
                </tr>
              </thead>
              <tbody>
                {data &&
                  data.List_products.map((values, index) => (
                    <tr key={index}>
                      <td>{index + 1}</td>
                      <td>{values.productname}</td>
                      <td>{values.productcategory}</td>
                      <td>{values.productimage}</td>
                      <td>{values.productfreshness}</td>
                      <td>{values.productdescription}</td>
                      <td>{values.productprice}$</td>
                      <td>
                        <ButtonDelete
                          onDelete={() => {
                            handleDeleteProduct(values.id);
                          }}
                        />
                        <ButtonEdit />
                      </td>
                    </tr>
                  ))}
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </>
  );
};

export default Form;
