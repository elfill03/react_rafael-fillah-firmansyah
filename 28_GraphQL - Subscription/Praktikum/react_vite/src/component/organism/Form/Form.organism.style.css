.navigation {
  border: none;
  margin-bottom: 20px;
  padding-top: 5px;
  padding-bottom: 5px;
  background-color: rgb(232, 250, 215);
}
.navigation:hover {
  background-color: rgb(196, 255, 140);
}
.error-msg{
  color: red;
  font-size: 16px;
  margin-top: 5px;
}
