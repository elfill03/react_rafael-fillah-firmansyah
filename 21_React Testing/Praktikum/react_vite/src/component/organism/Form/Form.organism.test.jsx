import renderer from "react-test-renderer";
import Form from "./Form.organism";

describe("Form.organism.jsx", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<Form />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
