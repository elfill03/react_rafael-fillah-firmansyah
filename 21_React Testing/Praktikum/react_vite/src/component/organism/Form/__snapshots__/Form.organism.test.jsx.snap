// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Form.organism.jsx > should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="d-grid"
  >
    <a
      className="navbar-brand d-flex justify-content-center mt-4"
      href="https://getbootstrap.com/"
    >
      <img
        alt="Bootstrap"
        height={44}
        src="/src/assets/svg/bootstraplogo.svg"
        width={50}
      />
    </a>
    <div
      className="text-center mt-4 mb-4"
    >
      <div
        className="d-flex text-center"
      >
        <h1
          className="fs-4 ms-auto mt-auto mb-auto "
        >
          Choose Language: 
        </h1>
        <button
          className="button-language me-auto"
          onClick={[Function]}
        >
          En
        </button>
      </div>
      <h1
        className="fs-2 fw-bold"
      >
        Create Product
      </h1>
      <h2
        className="fs-4 fw-lighter"
      >
        Below is an example form built entirely with Bootstrapâ€™s form controls. Each required form group has a validation state that can be triggered by attempting to submit the form without completing it.
      </h2>
    </div>
    <form
      autoComplete="off"
      className="needs-validation"
      onSubmit={[Function]}
    >
      <div
        className="d-grid m-auto has-validation"
        style={
          {
            "width": "min(100% - 10px, 40%)",
          }
        }
      >
        <button
          className="generate"
          onClick={[Function]}
        >
          Generate Number
        </button>
        <button
          className="navigation"
        >
          Move To Landing Page
        </button>
        <h1
          className="fs-3 fw-bold"
        >
          Detail Product
        </h1>
        <div>
          <label
            className="form-label"
            htmlFor="productname"
          >
            Product Name
          </label>
          <input
            aria-describedby="productname"
            autoFocus=""
            className="form-control"
            id="productname"
            name="productname"
            onChange={[Function]}
            style={
              {
                "width": "min(100% - 10px, 60%)",
              }
            }
            type="text"
            value=""
          />
        </div>
        <div
          className="has-validation mt-3"
        >
          <label
            className="form-label"
            htmlFor="productcategory"
          >
            Product Category :
          </label>
          <select
            className="form-select mb-3"
            name="productcategory"
            onChange={[Function]}
            style={
              {
                "width": "min(100% - 10px, 45%)",
              }
            }
            value=""
          >
            <option
              hidden=""
            >
              Choose...
            </option>
            <option>
              Food
            </option>
            <option>
              Fruit
            </option>
            <option>
              Vegetable
            </option>
          </select>
        </div>
        <div
          className="has-validation mt-3"
        >
          <label
            className="form-label"
            htmlFor="formFile"
          >
            Image of Product
          </label>
          <input
            className="form-control text-primary border border-primary"
            id="productimage"
            name="productimage"
            onChange={[Function]}
            style={
              {
                "width": "min(100% - 10px, 58%)",
              }
            }
            type="file"
            value=""
          />
        </div>
        <div
          className="has-validation mt-3"
        >
          <label
            className="form-label"
          >
            Product Freshness :
          </label>
          <div
            className="form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              id="freshness1"
              name="productfreshness"
              onChange={[Function]}
              type="radio"
              value="Brand New"
            />
            <label
              className="form-check-label"
              htmlFor="freshness1"
            >
              Brand New
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              id="freshness2"
              name="productfreshness"
              onChange={[Function]}
              type="radio"
              value="Second Hank"
            />
            <label
              className="form-check-label"
              htmlFor="freshness2"
            >
              Second Hank
            </label>
          </div>
          <div
            className="form-check"
          >
            <input
              checked={false}
              className="form-check-input"
              id="freshness3"
              name="productfreshness"
              onChange={[Function]}
              type="radio"
              value="Refurbished"
            />
            <label
              className="form-check-label"
              htmlFor="freshness3"
            >
              Refurbished
            </label>
          </div>
        </div>
        <div
          className="has-validation mt-3"
        >
          <label
            className="form-label"
            htmlFor="Description"
          >
            Additional Description
          </label>
          <textarea
            className="form-control"
            id="Description"
            name="productdescription"
            onChange={[Function]}
            rows={4}
            value=""
          />
        </div>
        <div
          className="has-validation mt-3"
        >
          <label
            className="form-label"
            htmlFor="productprice"
          >
            Product Price :
          </label>
          <input
            aria-describedby="productprice"
            className="form-control"
            id="productprice"
            name="productprice"
            onChange={[Function]}
            placeholder="$ 0"
            type="number"
            value={0}
          />
        </div>
        <button
          className="btn btn-primary m-auto mt-5 mb-5"
          id="submit"
          onClick={[Function]}
          style={
            {
              "width": "min(100% - 10px, 90%)",
            }
          }
          type="submit"
        >
          Create Product
        </button>
      </div>
    </form>
    <div
      className="container m-auto text-center"
    >
      <h1
        className="fw-bold fs-2"
      >
        List Product
      </h1>
    </div>
    <div
      className="container-fluid mb-0"
    >
      <table
        className="table table-light table-striped"
        id="tableData"
        style={
          {
            "width": "min(100% - 10px, 75%)",
          }
        }
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              No
            </th>
            <th
              scope="col"
            >
              Product Name
            </th>
            <th
              scope="col"
            >
              Product Category
            </th>
            <th
              scope="col"
            >
              Image of Product
            </th>
            <th
              scope="col"
            >
              Product Freshness
            </th>
            <th
              scope="col"
            >
              Additional Desciption
            </th>
            <th
              scope="col"
            >
              Product Price
            </th>
            <th
              scope="col"
            >
              Action
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              1
            </td>
            <td />
            <td />
            <td />
            <td />
            <td />
            <td>
              $
            </td>
            <td>
              <button
                className="Btn-delete"
                onClick={[Function]}
              >
                Delete
              </button>
              <button
                className="Btn-edit"
              >
                Edit
              </button>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
